lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1246,42 +1248,55 @@ def configure_hardware_params(self, vm_obj):
             if 'version' in self.params['hardware']:
                 hw_version_check_failed = False
                 temp_version = self.params['hardware'].get('version', 10)
-                try:
-                    temp_version = int(temp_version)
-                except ValueError:
-                    hw_version_check_failed = True
+                if temp_version.lower() == 'latest':
```

Comment:
There is already way too much if in this part of the code, may I suggest you split up the logic in a new function?